python/test/block/Test_ACEtk_KalbachMannDistributionData.py

```diff
@@ -0,0 +1,131 @@
+    """Unit test for the KalbachMannDistributionData class."""
+
+    chunk = [            0,             2,  1.219437E+01,  2.000000E+01,
+                        27,            39,             1,             2,
+              0.000000E+00,  1.866919E-02,  5.356419E+01,  0.000000E+00,
+              0.000000E+00,  1.000000E+00,  0.000000E+00,  0.000000E+00,
+              2.391154E-01,  2.398743E-01,             2,             3,
+              0.000000E+00,  1.120151E+00,  7.592137E+00,  7.738696E-02,
+              4.209016E-01,  1.226090E-11,  0.000000E+00,  5.382391E-01,
+              1.000000E+00,  2.491475E-03,  1.510768E-02,  9.775367E-01,
+              2.391154E-01,  2.847920E-01,  5.592013E-01 ]
+
+    def test_component( self ) :
+
+        def verify_chunk( self, chunk ) :
+
+            # verify content
+            self.assertEqual( False, chunk.empty )
+            self.assertEqual( 35, chunk.length )
+            self.assertEqual( "DLW::KalbachMannDistributionData", chunk.name )
+
+            self.assertEqual( 2, chunk.NE )
+            self.assertEqual( 2, chunk.number_incident_energies )
+
+            self.assertEqual( 2, len( chunk.incident_energies ) )
+            self.assertAlmostEqual( 1.219437E+01, chunk.incident_energies[0] )
+            self.assertAlmostEqual( 20., chunk.incident_energies[1] )
+
+            self.assertAlmostEqual( 1.219437E+01, chunk.incident_energy(1) )
+            self.assertAlmostEqual( 20., chunk.incident_energy(2) )
+
+            self.assertEqual( 27, chunk.LOCC(1) );
+            self.assertEqual( 39, chunk.LOCC(2) );
+            self.assertEqual( 27, chunk.distribution_locator(1) );
+            self.assertEqual( 39, chunk.distribution_locator(2) );
+
+            self.assertEqual( 7, chunk.relative_distribution_locator(1) );
+            self.assertEqual( 19, chunk.relative_distribution_locator(2) );
+
+            self.assertEqual( True, isinstance( chunk.distribution(1), TabulatedKalbachMannDistribution )  )
+            self.assertEqual( True, isinstance( chunk.distribution(2), TabulatedKalbachMannDistribution ) )
+
+            data1 = chunk.distribution(1);
+            self.assertAlmostEqual( 1.219437E+01, data1.incident_energy )
+            self.assertEqual( 1, data1.interpolation )
+            self.assertEqual( 0, data1.number_discrete_photon_lines )
+            self.assertEqual( 2, data1.number_outgoing_energies )
+
+            self.assertEqual( 2, len( data1.outgoing_energies ) )
+            self.assertAlmostEqual( 0., data1.outgoing_energies[0] )
+            self.assertAlmostEqual( 1.866919E-02, data1.outgoing_energies[-1] )
+
+            self.assertEqual( 2, len( data1.pdf ) )
+            self.assertAlmostEqual( 5.356419E+01, data1.pdf[0] )
+            self.assertAlmostEqual( 0., data1.pdf[-1] )
+
+            self.assertEqual( 2, len( data1.cdf ) )
+            self.assertAlmostEqual( 0., data1.cdf[0] )
+            self.assertAlmostEqual( 1., data1.cdf[-1] )
```

Comment:
There are spots where we have small differences, yes. Whether those are representation issues - or read issues in ACEtk I do not know. Tim did tell me that the disco routines we use to read the ASCII data are always off by a single bit. This is why 1e-11 sometimes comes out as 9.9999999...e-12 instead of 1e-11 exactly. Using strtod() instead of disco seems to fix that. It is an issue I will have to look into at some point.